api/sip/client.py

```diff
@@ -174,14 +183,16 @@ class SIPClient(Constants):
     log = logging.getLogger("SIPClient")
    
     def __init__(self, target_server, target_port, login_user_id=None,
-                 login_password=None, separator='|'):
+                 login_password=None, location_code=None, separator=None):
         self.target_server = target_server
         if target_port:
             self.target_port = int(target_port)
+        self.location_code = location_code
+        self.separator = separator or '|'
```

Comment:
When SIP2AuthenticationProvider is instantiated from config, if no field separator is provided in the config, the value is set to None. Since a value is available for field_separator, even though the value is None, the SIPClient constructor was being called with separator=None. This code makes means that you get the default separator whether you pass in separator=None or omit the separator altogether.